src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerImpl.java

```diff
@@ -129,5 +155,42 @@ private void registerWebhook(BitbucketSCMRepository repository) {
             BitbucketWebhook webhook = retryingWebhookHandler.register(repository);
             LOGGER.info("Webhook returned -" + webhook);
         }
+
+        private boolean isWebhookExists(Job<?, ?> project, BitbucketSCM input) {
+            try (ACLContext ctx = ACL.as(ACL.SYSTEM)) {
+                return jenkinsProvider
+                        .get().getAllItems(ParameterizedJobMixIn.ParameterizedJob.class)
```

Comment:
Is this too expensive to do on every pipelines build? I think that when a pipelines build gets triggered that this will be called. I don't know any other way of doing it, but we're also going to do it another time in the Webhook Consumer so maybe there's a way to dedupe this work?

Reply:
I don't think it is expensive. It is an in-memory operation with max iteration over hundreds of item not even thousands. I don't think we should optimize without measuring.